File: PP_02.01/DrivingSchoolWebAPI2/DrivingSchoolAPIModels/ApiModels/InstructorScheduleModel.cs
using System.ComponentModel.DataAnnotations;

namespace DrivingSchoolAPIModels
{
    /// <summary>
    /// Модель для передачи данных расписания инструктора на сервер
    /// </summary>
    public class InstructorScheduleModel
    {
        [Required]
        public int InstructorId { get; set; }
        [Required]
        public DateOnly DayOfWork { get; set; }
        [Required]
        public ClassModel[] Classes { get; set; }
    }
}

File: PDP/DrivingSchoolAppAndroidClient/app/src/main/java/com/example/drivingschoolappandroidclient/api/models/InstructorScheduleModel.kt
package com.example.drivingschoolappandroidclient.api.models

import com.example.drivingschoolappandroidclient.App
import java.sql.Date
import java.time.LocalDate

/**
 * Модель для передачи данных расписания инструктора на сервер
 */
data class InstructorScheduleModel(
    var instructorId: Int,
    var dayOfWorkJson: String,
    var classes: Array<ClassModel>,
){
    var dayOfWork: LocalDate
        get() = App.dateFromString(dayOfWorkJson)
        set(value) {
            dayOfWorkJson = App.dateToString(value)
        }
}
